### PR TITLE
(refactor): partial fix for issue #763 by refactoring common e2e code

### PR DESCRIPTION
**Pull request by maximilien**
## Description

Addresses the `e2e` part of issue #763. Moves common code to `lib\test\integration`

## Changes

* refactors the common `e2e` code into `lib\test\integration`
* change package name of the common code from `e2e` to integration
* change all `e2e` test to use new common code

## Reference

Partially fix to issue #763

### CODE REVIEW

File: lib/test/integration/flags.go [hunk 1]
```diff
@@ -23,19 +23,20 @@ import (
...
+func InitializeFlags() *ClientFlags {
 	var f ClientFlags
 
 	dockerConfigJSON := os.Getenv("DOCKER_CONFIG_JSON")
 	flag.StringVar(&f.DockerConfigJSON, "dockerconfigjson", dockerConfigJSON,
 		"Provide the path to Docker configuration file in json format. Defaults to $DOCKER_CONFIG_JSON")
 
 	return &f
-}
+}
```
Comment (mattmoor) [inline lib/test/integration/flags.go]: Add trailing newline:
```suggestion
}

```

File: lib/test/integration/result_collector.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+// KnRunResultCollector collects Kn run's results
+type KnRunResultCollector struct {
+	results    []KnRunResult
+	extraDumps []string
+	t          *testing.T
+}
+
+// NewKnRunResultCollector
```
Comment (mattmoor) [inline lib/test/integration/result_collector.go]: Remove trailing whitespace:
```suggestion
// NewKnRunResultCollector
```

File: test/e2e/basic_workflow_test.go [hunk 3]
```diff
@@ -136,10 +137,10 @@ func (test *e2eTest) revisionListForService(t *testing.T, r *KnRunResultCollecto
...
+func revisionDescribe(t *testing.T, it *integration.Test, r *integration.KnRunResultCollector, serviceName string) {
+	revName := findRevision(t, it, r, serviceName)
 
-	out := test.kn.Run("revision", "describe", revName)
+	out := it.Kn().Run("revision", "describe", revName)
 	r.AssertNoError(out)
-	assert.Check(t, util.ContainsAll(out.Stdout, revName, test.kn.namespace, serviceName, "++ Ready", "TARGET=kn"))
-}
+	assert.Check(t, util.ContainsAll(out.Stdout, revName, it.Kn().Namespace(), serviceName, "++ Ready", "TARGET=kn"))
+}
```
Comment (mattmoor) [inline test/e2e/basic_workflow_test.go]: Add trailing newline:
```suggestion
}

```

File: lib/test/integration/flags.go [hunk 4]
```diff
@@ -23,19 +23,20 @@ import (
...
-func initializeFlags() *ClientFlags {
+// InitializeFlags initializes the client's flags
+func InitializeFlags() *ClientFlags {
 	var f ClientFlags
 
 	dockerConfigJSON := os.Getenv("DOCKER_CONFIG_JSON")
 	flag.StringVar(&f.DockerConfigJSON, "dockerconfigjson", dockerConfigJSON,
 		"Provide the path to Docker configuration file in json format. Defaults to $DOCKER_CONFIG_JSON")
 
 	return &f
```
Comment (mattmoor) [inline lib/test/integration/flags.go]: Format Go code:
```suggestion
	return &f
}
```

File: lib/test/integration/integration.go [hunk 5]
```diff
@@ -38,60 +38,69 @@ var serviceMutex sync.Mutex
...
 
-func getNextNamespaceId() int {
+// GetNextNamespaceId return the next unique ID for the next namespace
+func GetNextNamespaceId() int {
 	nsMutex.Lock()
 	defer nsMutex.Unlock()
 	current := namespaceCount
 	namespaceCount++
 	return current
 }
```
Comment (mattmoor) [inline lib/test/integration/integration.go]: Format Go code:
```suggestion
}

```

File: lib/test/integration/result_collector.go [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"testing"
```
Comment (mattmoor) [inline lib/test/integration/result_collector.go]: Format Go code:
```suggestion

```

File: lib/test/integration/result_collector.go [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+	"strings"
```
Comment (mattmoor) [inline lib/test/integration/result_collector.go]: Format Go code:
```suggestion
	"strings"
	"testing"
```

File: lib/test/integration/result_collector.go (hunk 2 above)
Comment (mattmoor) [inline lib/test/integration/result_collector.go]: Format Go code:
```suggestion
// NewKnRunResultCollector
```

File: test/e2e/basic_workflow_test.go [hunk 8]
```diff
@@ -136,10 +137,10 @@ func (test *e2eTest) revisionListForService(t *testing.T, r *KnRunResultCollecto
...
-	revName := test.findRevision(t, r, serviceName)
+func revisionDescribe(t *testing.T, it *integration.Test, r *integration.KnRunResultCollector, serviceName string) {
+	revName := findRevision(t, it, r, serviceName)
 
-	out := test.kn.Run("revision", "describe", revName)
+	out := it.Kn().Run("revision", "describe", revName)
 	r.AssertNoError(out)
-	assert.Check(t, util.ContainsAll(out.Stdout, revName, test.kn.namespace, serviceName, "++ Ready", "TARGET=kn"))
-}
+	assert.Check(t, util.ContainsAll(out.Stdout, revName, it.Kn().Namespace(), serviceName, "++ Ready", "TARGET=kn"))
```
Comment (mattmoor) [inline test/e2e/basic_workflow_test.go]: Format Go code:
```suggestion
	assert.Check(t, util.ContainsAll(out.Stdout, revName, it.Kn().Namespace(), serviceName, "++ Ready", "TARGET=kn"))
}
```

File: test/e2e/plugins_test.go [hunk 9]
```diff
@@ -26,6 +26,7 @@ import (
 	"gotest.tools/assert"
 
 	"knative.dev/client/pkg/util"
```
Comment (mattmoor) [inline test/e2e/plugins_test.go]: Format Go code:
```suggestion

```

File: test/e2e/plugins_test.go [hunk 10]
```diff
@@ -26,6 +26,7 @@ import (
...
+	"knative.dev/client/lib/test/integration"
```
Comment (mattmoor) [inline test/e2e/plugins_test.go]: Format Go code:
```suggestion
	"knative.dev/client/lib/test/integration"
	"knative.dev/client/pkg/util"
```

File: test/e2e/sinkprefix_test.go [hunk 11]
```diff
@@ -22,7 +22,9 @@ import (
 	"testing"
 
 	"gotest.tools/assert"
+
 	"knative.dev/client/pkg/util"
```
Comment (mattmoor) [inline test/e2e/sinkprefix_test.go]: Format Go code:
```suggestion

```

File: test/e2e/sinkprefix_test.go [hunk 12]
```diff
@@ -22,7 +22,9 @@ import (
...
+	"knative.dev/client/lib/test/integration"
```
Comment (mattmoor) [inline test/e2e/sinkprefix_test.go]: Format Go code:
```suggestion
	"knative.dev/client/lib/test/integration"
	"knative.dev/client/pkg/util"
```

File: test/e2e/source_apiserver_test.go [hunk 13]
```diff
@@ -37,117 +38,116 @@ const (
...
-		"--serviceaccount="+test.kn.namespace+":"+testServiceAccount)
+		clusterRoleBindingPrefix+it.Kn().Namespace(),
+		"--clusterrole="+clusterRolePrefix+it.Kn().Namespace(),
+		"--serviceaccount="+it.Kn().Namespace()+":"+testServiceAccount)
 	assert.NilError(t, err)
 }
 
-func (test *e2eTest) tearDownForSourceApiServer() error {
-
+func tearDownForSourceApiServer(t *testing.T, it *integration.Test,) error {
```
Comment (mattmoor) [inline test/e2e/source_apiserver_test.go]: Format Go code:
```suggestion
func tearDownForSourceApiServer(t *testing.T, it *integration.Test) error {
```

File: test/e2e/source_list_test.go [hunk 14]
```diff
@@ -23,58 +23,59 @@ import (
 	"gotest.tools/assert"
 
 	"knative.dev/client/pkg/util"
```
Comment (mattmoor) [inline test/e2e/source_list_test.go]: Format Go code:
```suggestion

```

File: test/e2e/source_list_test.go [hunk 15]
```diff
@@ -23,58 +23,59 @@ import (
...
+	"knative.dev/client/lib/test/integration"
```
Comment (mattmoor) [inline test/e2e/source_list_test.go]: Format Go code:
```suggestion
	"knative.dev/client/lib/test/integration"
	"knative.dev/client/pkg/util"
```

File: lib/test/integration/result_collector.go (hunk 2 above)
Comment (mattmoor) [inline lib/test/integration/result_collector.go]: Remove trailing whitespace:
```suggestion
// NewKnRunResultCollector
```

File: test/e2e/basic_workflow_test.go (hunk 3 above)
Comment (mattmoor) [inline test/e2e/basic_workflow_test.go]: Add trailing newline:
```suggestion
}

```

**Comment by maximilien** [comment]
Closing this so I can resubmit. Forgot to `gofmt` new `lib` directory

**Pull request closed without merging**